diff -bru raddb.orig/clients.conf raddb/clients.conf
--- raddb.orig/clients.conf	2018-03-16 14:57:14.893362000 +0100
+++ raddb/clients.conf	2018-03-21 13:23:41.513395000 +0100
@@ -17,7 +17,27 @@
 #  that you delete, or comment out, this entry.
 #
 #
+
+client arubas {
+	ipv4addr = 10.3.4.0/24
+	secret = Secret_shared_with_arubas
+	shortname = arubas
+}
 
 #
 #  Each client has a "short name" that is used to distinguish it from
 #  other clients.
@@ -217,10 +237,10 @@
 }
 
 # IPv6 Client
-client localhost_ipv6 {
-	ipv6addr	= ::1
-	secret		= testing123
-}
+#client localhost_ipv6 {
+#	ipv6addr	= ::1
+#	secret		= testing123
+#}
 
 # All IPv6 Site-local clients
 #client sitelocal_ipv6 {
diff -bru raddb.orig/dictionary raddb/dictionary
--- raddb.orig/dictionary	2018-03-16 14:57:14.903438000 +0100
+++ raddb/dictionary	2018-03-21 14:43:10.910719000 +0100
@@ -47,3 +47,6 @@
 #ATTRIBUTE	My-Local-String		3000	string
 #ATTRIBUTE	My-Local-IPAddr		3001	ipaddr
 #ATTRIBUTE	My-Local-Integer	3002	integer
+ATTRIBUTE Expire-After 3000 integer
+ATTRIBUTE First-Login 3001 date
+
diff -bru raddb.orig/mods-available/sql raddb/mods-available/sql
--- raddb.orig/mods-available/sql	2018-03-16 14:57:14.898161000 +0100
+++ raddb/mods-available/sql	2018-03-21 12:41:01.837902000 +0100
@@ -28,7 +28,7 @@
 	#    * rlm_sql_sqlite
 	#    * rlm_sql_null (log queries to disk)
 	#
-	driver = "rlm_sql_null"
+	driver = "rlm_sql_postgresql"
 
 #
 #	Several drivers accept specific options, to set them, a
@@ -84,14 +84,13 @@
 	#
 	# If you're using rlm_sql_null, then it should be the type of
 	# database the logged queries are going to be executed against.
-	dialect = "sqlite"
+	dialect = "postgresql"
 
 	# Connection info:
 	#
-#	server = "localhost"
-#	port = 3306
-#	login = "radius"
-#	password = "radpass"
+	server = "our.db.server"
+	login = "radius"
+	password = "radius_db_password"
 
 	# Database table configuration for everything except Oracle
 	radius_db = "radius"
diff -bru raddb.orig/mods-config/sql/main/postgresql/queries.conf raddb/mods-config/sql/main/postgresql/queries.conf
--- raddb.orig/mods-config/sql/main/postgresql/queries.conf	2018-03-16 14:57:14.883025000 +0100
+++ raddb/mods-config/sql/main/postgresql/queries.conf	2018-03-21 14:32:07.755454000 +0100
@@ -98,6 +98,17 @@
 #
 #  Use these for case insensitive usernames. WARNING: Slower queries!
 #
+authorize_check_query = "SELECT id, UserName, Attribute, Value, Op \
+  FROM ${authcheck_table} \
+  WHERE Username = '%{SQL-User-Name}' \
+  ORDER BY id"
+#  AND ( SELECT extract ( epoch from ( to_timestamp(value, 'Mon DD YYYY HH24:MI'))) FROM radcheck where Username = '%{User-Name}' and Attribute = 'First-Login' ) > ( SELECT extract ( epoch from now()) :: bigint ) \
+
+authorize_reply_query = "SELECT id, UserName, Attribute, Value, Op \
+  FROM ${authreply_table} \
+  WHERE Username = '%{SQL-User-Name}' \
+  ORDER BY id"
+
 #authorize_check_query = "\
 #	SELECT id, UserName, Attribute, Value, Op \
 #	FROM ${authcheck_table} \
@@ -114,6 +125,7 @@
 	SELECT id, UserName, Attribute, Value, Op \
 	FROM ${authcheck_table} \
 	WHERE Username = '%{SQL-User-Name}' \
+ 	AND ( SELECT extract ( epoch from ( to_timestamp(value, 'Mon DD YYYY HH24:MI'))) FROM radcheck where Username = '%{User-Name}' and Attribute = 'First-Login' ) > ( SELECT extract ( epoch from now()) :: bigint ) \
 	ORDER BY id"
 
 authorize_reply_query = "\
Only in raddb.orig/mods-enabled: cache_eap
Only in raddb.orig/mods-enabled: eap
Only in raddb/mods-enabled: sql
Only in raddb.orig/mods-enabled: sradutmp
Only in raddb/mods-enabled: ssqlcounter
Only in raddb.orig/mods-enabled: unix
diff -bru raddb.orig/radiusd.conf raddb/radiusd.conf
--- raddb.orig/radiusd.conf	2018-03-16 14:57:14.939191000 +0100
+++ raddb/radiusd.conf	2018-03-16 15:02:09.931458000 +0100
@@ -398,8 +398,8 @@
 	#  member.  This can allow for some finer-grained access
 	#  controls.
 	#
-#	user = radius
-#	group = radius
+	user = freeradius
+	group = freeradius
 
 	#  Core dumps are a bad thing.  This should only be set to
 	#  'yes' if you're debugging a problem with the server.
@@ -480,8 +480,8 @@
 #
 #  allowed values: {no, yes}
 #
-proxy_requests  = yes
-$INCLUDE proxy.conf
+proxy_requests  = no
+#$INCLUDE proxy.conf
 
 
 # CLIENTS CONFIGURATION
diff -bru raddb.orig/sites-available/default raddb/sites-available/default
--- raddb.orig/sites-available/default	2018-03-16 14:57:14.891584000 +0100
+++ raddb/sites-available/default	2018-03-21 14:44:32.750421000 +0100
@@ -231,33 +231,33 @@
 }
 
 # IPv6 versions of the above - read their full config to understand options
-listen {
-	type = auth
-	ipv6addr = ::	# any.  ::1 == localhost
-	port = 0
-#	interface = eth0
-#	clients = per_socket_clients
-	limit {
-	      max_connections = 16
-	      lifetime = 0
-	      idle_timeout = 30
-	}
-}
-
-listen {
-	ipv6addr = ::
-	port = 0
-	type = acct
-#	interface = eth0
-#	clients = per_socket_clients
-
-	limit {
-#		max_pps = 0
-#		idle_timeout = 0
+#listen {
+#	type = auth
+#	ipv6addr = ::	# any.  ::1 == localhost
+#	port = 1814
+##	interface = eth0
+##	clients = per_socket_clients
+#	limit {
+#	      max_connections = 16
 #		lifetime = 0
-#		max_connections = 0
-	}
-}
+#	      idle_timeout = 30
+#	}
+#}
+#
+#listen {
+#	ipv6addr = ::
+#	port = 1815
+#	type = acct
+##	interface = eth0
+##	clients = per_socket_clients
+#
+#	limit {
+##		max_pps = 0
+##		idle_timeout = 0
+##		lifetime = 0
+##		max_connections = 0
+#	}
+#}
 
 #  Authorization. First preprocess (hints and huntgroups files),
 #  then realms, and finally look in the "users" file.
@@ -379,10 +379,10 @@
 	#  uncomment it as well; this will further reduce the number of
 	#  LDAP and/or SQL queries for TTLS or PEAP.
 	#
-	eap {
-		ok = return
-#		updated = return
-	}
+	#eap {
+	#	ok = return
+#	#	updated = return
+	#}
 
 	#
 	#  Pull crypt'd passwords from /etc/passwd or /etc/shadow,
@@ -529,7 +529,7 @@
 
 	#
 	#  Allow EAP authentication.
-	eap
+	#eap
 
 	#
 	#  The older configurations sent a number of attributes in
@@ -620,7 +620,7 @@
 	#  Update the wtmp file
 	#
 	#  If you don't use "radlast", you can delete this line.
-	unix
+	#unix
 
 	#
 	#  For Simultaneous-Use tracking.
@@ -679,7 +679,7 @@
 
 	#
 	#  See "Simultaneous Use Checking Queries" in mods-available/sql
-#	sql
+	sql
 }
 
 
@@ -815,7 +815,7 @@
 #	}
 
 	#  Remove reply message if the response contains an EAP-Message
-	remove_reply_message_if_eap
+	#remove_reply_message_if_eap
 
 	#
 	#  Access-Reject packets are sent through the REJECT sub-section of the
@@ -834,10 +834,10 @@
 		# Insert EAP-Failure message if the request was
 		# rejected by policy instead of because of an
 		# authentication failure
-		eap
+		#eap
 
 		#  Remove reply message if the response contains an EAP-Message
-		remove_reply_message_if_eap
+		#remove_reply_message_if_eap
 	}
 
 	#
@@ -912,7 +912,7 @@
 	#  hidden inside of the EAP packet, and the end server will
 	#  reject the EAP request.
 	#
-	eap
+	#eap
 
 	#
 	#  If the server tries to proxy a request and fails, then the
diff -bru raddb.orig/sites-enabled/default raddb/sites-enabled/default
--- raddb.orig/sites-enabled/default	2018-03-16 14:57:14.891584000 +0100
+++ raddb/sites-enabled/default	2018-03-21 14:44:32.750421000 +0100
@@ -231,33 +231,33 @@
 }
 
 # IPv6 versions of the above - read their full config to understand options
-listen {
-	type = auth
-	ipv6addr = ::	# any.  ::1 == localhost
-	port = 0
-#	interface = eth0
-#	clients = per_socket_clients
-	limit {
-	      max_connections = 16
-	      lifetime = 0
-	      idle_timeout = 30
-	}
-}
-
-listen {
-	ipv6addr = ::
-	port = 0
-	type = acct
-#	interface = eth0
-#	clients = per_socket_clients
-
-	limit {
-#		max_pps = 0
-#		idle_timeout = 0
+#listen {
+#	type = auth
+#	ipv6addr = ::	# any.  ::1 == localhost
+#	port = 1814
+##	interface = eth0
+##	clients = per_socket_clients
+#	limit {
+#	      max_connections = 16
 #		lifetime = 0
-#		max_connections = 0
-	}
-}
+#	      idle_timeout = 30
+#	}
+#}
+#
+#listen {
+#	ipv6addr = ::
+#	port = 1815
+#	type = acct
+##	interface = eth0
+##	clients = per_socket_clients
+#
+#	limit {
+##		max_pps = 0
+##		idle_timeout = 0
+##		lifetime = 0
+##		max_connections = 0
+#	}
+#}
 
 #  Authorization. First preprocess (hints and huntgroups files),
 #  then realms, and finally look in the "users" file.
@@ -379,10 +379,10 @@
 	#  uncomment it as well; this will further reduce the number of
 	#  LDAP and/or SQL queries for TTLS or PEAP.
 	#
-	eap {
-		ok = return
-#		updated = return
-	}
+	#eap {
+	#	ok = return
+#	#	updated = return
+	#}
 
 	#
 	#  Pull crypt'd passwords from /etc/passwd or /etc/shadow,
@@ -529,7 +529,7 @@
 
 	#
 	#  Allow EAP authentication.
-	eap
+	#eap
 
 	#
 	#  The older configurations sent a number of attributes in
@@ -620,7 +620,7 @@
 	#  Update the wtmp file
 	#
 	#  If you don't use "radlast", you can delete this line.
-	unix
+	#unix
 
 	#
 	#  For Simultaneous-Use tracking.
@@ -679,7 +679,7 @@
 
 	#
 	#  See "Simultaneous Use Checking Queries" in mods-available/sql
-#	sql
+	sql
 }
 
 
@@ -815,7 +815,7 @@
 #	}
 
 	#  Remove reply message if the response contains an EAP-Message
-	remove_reply_message_if_eap
+	#remove_reply_message_if_eap
 
 	#
 	#  Access-Reject packets are sent through the REJECT sub-section of the
@@ -834,10 +834,10 @@
 		# Insert EAP-Failure message if the request was
 		# rejected by policy instead of because of an
 		# authentication failure
-		eap
+		#eap
 
 		#  Remove reply message if the response contains an EAP-Message
-		remove_reply_message_if_eap
+		#remove_reply_message_if_eap
 	}
 
 	#
@@ -912,7 +912,7 @@
 	#  hidden inside of the EAP packet, and the end server will
 	#  reject the EAP request.
 	#
-	eap
+	#eap
 
 	#
 	#  If the server tries to proxy a request and fails, then the
Only in raddb.orig/sites-enabled: inner-tunnel
